### DISCUSSION TITLE
Can we increase the expiration time of ipam token

### DISCUSSION THREAD

**Discussion by SaumyaBhushan** [Azure/ipam#140] [General]
Currently Ipam tokens get expired in a very short span (within few hours). Can I define custom expiration time for ipam tokens?

**Comment by DCMattyG** [comment]
Hey @SaumyaBhushan, the Azure IPAM solution doesn't generate tokens, we simply leverage Azure AD tokens. There are mechanisms in place to change the lifetimes of AAD tokens, and you can find additional details on those options here:

https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes

I hope that helps!

**Reply by DCMattyG** [reply]
I'd like to point out that the tokens expire in a short period of time by design. Once a token is generated, it's valid until it expires or is revoked, so if someone were to get a hold of an AAD token, they would be able to do whatever that token allowed them to do until it's expiration. Extending the lifetime of a token can compromise security, so please make sure you understand the implications before doing so.

If you have a workflow that you're trying to implement that is being impacted by token expiration, I'm more than happy to offer some advise on alternative mechanisms that may work better for you and perhaps solve your problem in a different way. We're here to help!

**Reply by SaumyaBhushan** [reply]
Yes @DCMattyG  I am trying to implement a workflow . What are those alternative mechanisms ? Can you please guide me through that ?

**Reply by DCMattyG** [reply]
I'm happy to help @SaumyaBhushan, but any solution I would provide would be based on your specific circumstance. What are you trying to do here the token expiration is causing a pain-point? Is this something like Azure DevOps, or perhaps scripting like PowerShell? Maybe Terraform?

Help me understand a bit more about what you're trying to do, and I'll do my best to guide you through some more effective way to do it that can take into account token expiration times.

**Reply by SaumyaBhushan** [reply]
Yes Sure @DCMattyG ! So I am using azure devops pipeline to run terraform code for ipam reservation . There I am using token, so currently I am using az commands to get the tokens .

**Reply by DCMattyG** [reply]
Got it @SaumyaBhushan! And what are you doing that is taking so long that the token expires? Tokens are usually good for 1 hour when issued....is something in your Pipeline taking longer than an hour?

Source: https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes#saml-tokens

> ...The default lifetime of the token is 1 hour.

**Reply by SaumyaBhushan** [reply]
Hey @DCMattyG That answers my doubts . I am really thankful . These document links also helped me a lot to clear my doubts . I think I am good now .

**Reply by DCMattyG** [reply]
Excellent @SaumyaBhushan, so happy I could help. Please let me know if there's anything else you need!

**Reply by SaumyaBhushan** [reply]
No @DCMattyG . Thats all for now . Thank you !
